### PR TITLE
Doxygen Warnings

### PR DESCRIPTION
**Pull request by aul12**
Fixed all warnings by doxygen (i.e. fully documented all functions).

Relevant flags for reference:
```
WARN_IF_UNDOCUMENTED   = YES
WARN_IF_DOC_ERROR      = YES
WARN_NO_PARAMDOC       = YES
WARN_AS_ERROR          = YES
```

### CODE REVIEW

**Comment by aul12** [comment]
It would probably help to add the checks to the CI as well (this could also be combined with deploying the HTML-documentation to pages). Any opinions?

File: include/SI/SiImpl.hpp [hunk 1]
```diff
@@ -37,102 +61,259 @@ namespace si {
...
-        // Subtract (only same units)
+        /**
+         * Subtract object with SI object of same unit and type
+         * @param rhs the object to subtract
+         * @return the same of both values with same unit
+         */
         constexpr auto operator-(ThisT rhs) const -> ThisT;
 
+        /**
+         * Decrement Increment object by SI object of same unit and type
```
Comment (ottojo) [inline include/SI/SiImpl.hpp]:
```suggestion
         * Decrement object by SI object of same unit and type
```

**Comment by ottojo** [comment]
> It would probably help to add the checks to the CI as well (this could also be combined with deploying the HTML-documentation to pages). Any opinions?

Sounds good, can be done in separate/later PR IMO

**Comment by aul12** [comment]
@ottojo ping

**Review by ottojo — APPROVED** [review]

**Pull request merged**